clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -5791,12 +5783,32 @@ void SemaSYCL::MarkDevices() {
   }
 }
 
+static bool CheckFreeFunctionDiagnostics(Sema &S, FunctionDecl *FD) {
+  if (FD->isVariadic()) {
+    S.Diag(FD->getLocation(), diag::err_free_function_variadic_args);
+    return false;
```

Comment:
In clang it is often vice-versa, true is returned on an error. Because this way, you can do
```suggestion
    return S.Diag(FD->getLocation(), diag::err_free_function_variadic_args);
```